src/adapter/src/coord.rs

```diff
@@ -154,8 +154,11 @@ mod sequencer;
 mod sql;
 
 /// The default is set to a second to track the default timestamp frequency for sources.
-pub const DEFAULT_LOGICAL_COMPACTION_WINDOW_MS: Option<mz_repr::Timestamp> =
-    Some(Timestamp::new(1_000));
+pub const DEFAULT_LOGICAL_COMPACTION_WINDOW: Duration = Duration::from_secs(1);
+
+/// `DEFAULT_LOGICAL_COMPACTION_WINDOW` in the `EpochMillis` domain
+pub const DEFAULT_LOGICAL_COMPACTION_WINDOW_MS: mz_repr::Timestamp =
+    Timestamp::new(DEFAULT_LOGICAL_COMPACTION_WINDOW.as_millis() as u64);
```

Comment:
We're trying to avoid using `as`. Is this a `u128` to `u64` conversion? If so I think a `try_from`/`try_into` followed by an `expect` would be better.

Reply:
See discussion [here](https://materializeinc.slack.com/archives/CMH6PG4CW/p1671480553139239).

It's a u128 -> u64, but I can't use `try_from` because that's not a const function.

Once Petros lands some changes to `ore`, I'll be able to do this without `as`.

Reply:
Actually, there's a way to do it without `as` now, if I just factor out the integer millis value (see latest commit)